commercetools-models/src/main/java/io/sphere/sdk/customers/commands/CustomerSignInCommand.java

```diff
@@ -37,20 +39,27 @@
 public final class CustomerSignInCommand extends CommandImpl<CustomerSignInResult> {
     private final String email;
     private final String password;
+    /**
+     * @deprecated use {@link CustomerSignInCommand#anonymousCart} instead
+     */
+    @Deprecated
     @Nullable
     private final String anonymousCartId;
     @Nullable
+    private final ResourceIdentifier<Cart> anonymousCart;
+    @Nullable
     private final String anonymousId;
     @Nullable
     private final AnonymousCartSignInMode anonymousCartSignInMode;
     @Nullable
     private final Boolean updateProductData;
 
-    private CustomerSignInCommand(final String email, final String password, @Nullable final String anonymousCartId, @Nullable final String anonymousId, final AnonymousCartSignInMode anonymousCartSignInMode,
+    private CustomerSignInCommand(final String email, final String password, @Nullable final String anonymousCartId, @Nullable ResourceIdentifier<Cart> anonymousCart, @Nullable final String anonymousId, final AnonymousCartSignInMode anonymousCartSignInMode,
```

Comment:
same here. Drop the anonymousCartId and refactor the other methods to use create a ResourceIdentifier if needed

```suggestion
    private CustomerSignInCommand(final String email, final String password, @Nullable ResourceIdentifier<Cart> anonymousCart, @Nullable final String anonymousId, final AnonymousCartSignInMode anonymousCartSignInMode,
```